### PR TITLE
add feature:   entity_key_from_object_with_seed , entity_key_from_object_with_u256

### PR DESCRIPTION
**Pull request by v1xingyue**
add feature:  
1. entity_key_from_object_with_seed 
2. entity_key_from_object_with_u256

test method: 

base address to test : 0xa8e7fa2a93fd62713bfea9bfe47b59ab4590751972badc32d723f7d88a1fd804 
message seed :  "hello creator world" , this shuld get new address : 
 0xe87a6fbda67e3b539cacc7b9d3fa341f2b05cb7d784870d9501e53e611856d3f

u256 seed to test : 123456 , this should get new address : 0x90bb3d3d87b3d781b71d1b2b94644206293646e0e3fd477e71e4b0a8f20abe94

### CODE REVIEW

**Comment by web3olalala** [comment]
@v1xingyue I don't think these two methods are generic enough, can you give an example of a situation where they would be used. If it's a special case suggest a custom implementation in entity_key.move, here's an example from Constantinople.
https://github.com/0xobelisk/constantinople/blob/main/contracts/constantinople/sources/entity_key.move#L20

**Comment by v1xingyue** [comment]
> @v1xingyue I don't think these two methods are generic enough, can you give an example of a situation where they would be used. If it's a special case suggest a custom implementation in entity_key.move, here's an example from Constantinople. https://github.com/0xobelisk/constantinople/blob/main/contracts/constantinople/sources/entity_key.move#L20

Ok. 

 For example I have such a game.
1. Players can have weapons with diffrent names.  Each player can have a blockchain address as entity key as the player info. Then I can add a weapon name with the player entity key get another address as the entity key of the weapon belong to the player.
2.  Also as above example , I can have many weapons from 1 ~ 10. I can add number 1 to the weapon entity key get the indexed weapon .

**Comment by web3olalala** [comment]
I have a suggestion for this case, you can actually design the relationship between the player and the weapon this way.
image

I think it's enough to use directly from_bytes()/from_u256() to generate the weapon's entity ID, most of the time to let the player to perform the attack operation, it's always necessary to specify a weapon, such as `b "guns"`, `number`, so that it can easily get the weapon's entity ID, and then you can get some information about the weapon, and also you can use ` player_to_weapon` schema to check ownership.

**Comment by web3olalala** [comment]
Because `entity_key.move` needs to be abstract and concise enough, the functionality provided is limited, if you want to implement complex functionality it's best to do so through custom implementations, if things are added, most of them may actually be unneeded by the developer

**Comment by v1xingyue** [comment]
Also I want say,it also tell the realationship between weapon and my player address. 
It makes layered entity system. And also can be used some other places.
 It's a little like PDA in solana. 
It help derive address from one address with seed.

**Comment by v1xingyue** [comment]
Most important , In one game every player can have their own guns. 
Only u64 entity can't tell the owner ship between weapon and player. 
Then the game designers will need consider how to distinguish diffrent weapon ids between each other.

**Comment by web3olalala** [comment]
I understand, because entity_key.move is actually intended to be used as an example to the developer, because at present our entity_key only support address type, such as support for other types in the future, may delete the content, but for the time being has been the developer's needs are the main.

File: packages/common/src/codegen/utils/renderMove/generateEntityKey.ts [hunk 1]
```diff
@@ -18,6 +19,18 @@ export function generateEntityKey(config: ObeliskConfig, srcPrefix: string) {
     public fun from_u256(x: u256): address {
         address::from_u256(x)
     }
+
+    public fun from_object_with_seed(addr: address, seed: vector<u8>): address {
```
Comment (web3olalala) [inline packages/common/src/codegen/utils/renderMove/generateEntityKey.ts:23]: This should modify the method name from_address_with_seed.

File: packages/common/src/codegen/utils/renderMove/generateEntityKey.ts [hunk 2]
```diff
@@ -18,6 +19,18 @@ export function generateEntityKey(config: ObeliskConfig, srcPrefix: string) {
...
+        let data = address::to_bytes(addr);
+        vector::append(&mut data, seed);
+        from_bytes(data)
+    }
+
+    public fun from_object_with_u256(addr: address, x: u256): address {
```
Comment (web3olalala) [inline packages/common/src/codegen/utils/renderMove/generateEntityKey.ts:29]: This should modify the method name from_address_with_u256.

File: packages/common/src/codegen/utils/renderMove/generateEntityKey.ts [hunk 3]
```diff
@@ -18,6 +19,18 @@ export function generateEntityKey(config: ObeliskConfig, srcPrefix: string) {
...
+        let data = address::to_bytes(addr);
+        vector::append(&mut data, bcs::to_bytes<u256>(&x));
+        from_bytes(data)
+    }
```
Comment (web3olalala) [inline packages/common/src/codegen/utils/renderMove/generateEntityKey.ts:33]: Also need to add from_object_with_seed and from_object_with_u256

File: packages/common/src/codegen/utils/renderMove/generateEntityKey.ts (hunk 1 above)
Comment (web3olalala) [inline packages/common/src/codegen/utils/renderMove/generateEntityKey.ts]: This should modify the method name from_address_with_seed.

File: packages/common/src/codegen/utils/renderMove/generateEntityKey.ts (hunk 2 above)
Comment (web3olalala) [inline packages/common/src/codegen/utils/renderMove/generateEntityKey.ts]: This should modify the method name from_address_with_u256.

**Review by web3olalala — CHANGES_REQUESTED** [review]

**Comment by web3olalala** [comment]
Thank you for your contribution👍. @v1xingyue

**Pull request merged**